src/api/source/mvt.js

```diff
@@ -0,0 +1,452 @@
+                        numFields.push(c.name);
+                    } else if (type == 'category') {
+                        stringFields.push(c.name);
+                    } else {
+                        throw new Error(`Column type '${type}' not supported`);
+                    }
+
+                });
+                numFields.map((name, i) => fieldMap[name] = i);
+                jsonFields.map((name, i) => fieldMap[name] = i + numFields.length);
+                stringFields.map((name, i) => fieldMap[name] = i + jsonFields.length);
+
+                const { points, featureGeometries, properties, geomType } = this._decodeMVTLayer(mvtLayer, mvt_extent, this._MNS, jsonFields, numFields, stringFields, fieldMap);
+                this.metadata.geomType = geomType;
+
+                var rs = rsys.getRsysFromTile(x, y, z);
+                let dataframeProperties = {};
+                Object.keys(fieldMap).map((name, pid) => {
+                    dataframeProperties[name] = properties[pid];
+                });
+                let dataFrameGeometry = geomType == 'point' ? points : featureGeometries;
+                const dataframe = this._generateDataFrame(rs, dataFrameGeometry, dataframeProperties, mvtLayer.length, geomType);
+                this._addDataframe(dataframe);
+                return dataframe;
+            });
+    }
+
+    _getTileUrl(x, y, z) {
+        return this._serverURL.replace('{x}', x).replace('{y}', y).replace('{z}', z);
+    }
+
+    _getSubdomain(x, y) {
+        // Reference https://github.com/Leaflet/Leaflet/blob/v1.3.1/src/layer/tile/TileLayer.js#L214-L217
+        return this._subdomains[Math.abs(x + y) % this._subdomains.length];
+    }
+
+    _decodePolygons(geom, featureGeometries, mvt_extent) {
+        let polygon = null;
+        let geometry = [];
+        /*
+            All this clockwise non-sense is needed because the MVT decoder dont decode the MVT fully.
+            It doesn't distinguish between internal polygon rings (which defines holes) or external ones, which defines more polygons (mulipolygons)
+            See:
+                https://github.com/mapbox/vector-tile-spec/tree/master/2.1
+                https://en.wikipedia.org/wiki/Shoelace_formula
+        */
+        for (let j = 0; j < geom.length; j++) {
+            //if exterior
+            //   push current polygon & set new empty
+            //else=> add index to holes
+            let hole = false;
+            if (this._isClockWise(geom[j])) {
+                if (polygon) {
+                    geometry.push(polygon);
+                }
+                polygon = {
+                    flat: [],
+                    holes: [],
+                    clipped: [],
+                    clippedType: [], // Store a bitmask of the clipped half-planes
+                };
+            } else {
+                if (j == 0) {
+                    console.log('Invalid MVT tile: first polygon ring MUST be external');
```

Comment:
If this is the case, we may want to assume that the orientation is reversed entirely. In that case a `continue` won't be enough, but this depends much on our expectations regarding the backend.

Reply:
Here we have a situation. Some polygons are CW but because they have loops, for example, they are treated as CCW. Also, invalid polygons could fall into this category. Is something we have to work in. I'll give it a try after this is merged